streams/src/main/java/org/apache/kafka/streams/TopologyConfig.java

```diff
@@ -153,6 +166,17 @@ public TopologyConfig(final String topologyName, final StreamsConfig globalAppCo
         } else {
             deserializationExceptionHandlerSupplier = () -> globalAppConfigs.getConfiguredInstance(DEFAULT_DESERIALIZATION_EXCEPTION_HANDLER_CLASS_CONFIG, DeserializationExceptionHandler.class);
         }
+
+        // Override the default store type config no matter it's a named topology or not since it should apply to all topologies
+        storeType = getString(DEFAULT_DSL_STORE_CONFIG);
+        log.info("Topology {} is overriding {} to {}", topologyName, DEFAULT_DSL_STORE_CONFIG, storeType);
+    }
+
+    public Materialized.StoreType parseStoreType() {
+        if (storeType.equals(IN_MEMORY)) {
```

Comment:
nit: curious why `storeType` is not enum but raw string. Typo in `in_memory` could make it `ROCKS_DB` type. e.g. `in_memry`.

Reply:
@lihaosky , thanks for your comment. `storeType` is retrieved from stream config, where we can't accept enum. Currently, we only accept `rocksDB` or `in_memory`. If there's a typo, the config will throw exception, like we only accept `delete` and `compact` in `log.cleanup.policy`, and will throw exception if the provided value is other values.